README.md

```diff
@@ -108,17 +108,26 @@ own unit tests.
 
 ### Unit tests
 
+Unit tests within the project are built using [Mocha](https://mochajs.org/) and [Chai](https://www.chaijs.com/). Mocha is a test framework that provides the environment to write and run JavaScript tests, while Chai is an assertion library that allows us to write assertions in a more expressive and readable way.
+
+
 The following submodules contain unit tests: `core`, `crypto`, `circuits`,
 `contracts`, and `domainobjs`.
 
-Except for the `contracts` submodule, run unit tests as such (the following
-example is for `crypto`):
+You can run all unit tests from the root directory of the repo by running:
+
+```bash
+npm run test
+```
+
+Our you can run unit tests within each submodule. for example to run the `crypto` tests:
 
 ```bash
 cd crypto
 npm run test
 ```
 
+<!-- TODO: confirm if true -->
```

Comment:
@ctrlc03 is this necessary? Seems I'm able to run the full suite e.g. of `contracts` tests without nonce errors...

Reply:
> First, start a Hardhat instance in a separate terminal:
> ```bash
> cd contracts
> npm run hardhat
> ```

Is this needed? I'm able to just run `npm run test` within `contracts` & it seems to automatically start up Hardhat

Reply:
yea that's how it's supposed to work (hardhat does start itself) so yea feel free to remove. In general most of these README are outdated (wondering if it might better to address all in one PR?)

Reply:
> In general most of these README are outdated (wondering if it might better to address all in one PR?)

I'm _alright_ with that, though generally I think best to update documentation at the same time as the code is updated (ideally in the same PR), because:
- that tends to make it easier for PR creator, given the specific info & context is fresh in their mind
- it also makes it easier for the reviewer, so they can easily understand what's changing
- it avoids the issue of having this big chore for someone to pick up at the end of this refactoring effort. Not to mention if the chore drops as a priority, then everything is a mess 😄

Reply:
Noted! I'll make sure from now on all doc updates are included in the PR 🙂